executor/ddl.go

```diff
@@ -246,7 +246,11 @@ func (e *DDLExec) createSessionTemporaryTable(s *ast.CreateTableStmt) error {
 
 	// AutoID is allocated in mocked..
 	alloc := autoid.NewAllocatorFromTempTblInfo(tbInfo)
-	tbl, err := tables.TableFromMeta([]autoid.Allocator{alloc}, tbInfo)
+	allocs := make([]autoid.Allocator, 0, 1)
+	if alloc != nil {
+		allocs = append(allocs, alloc)
+	}
+	tbl, err := tables.TableFromMeta(allocs, tbInfo)
```

Comment:
why?

Reply:
we can know from memid.go L28 that NewAllocatorFromTempTblInfo will be return nil when there is not PK or COMMON PK and not exists auto_incr_id

Reply:
When `alloc` is nil, can the current code work?

Reply:
I mean, the read and write operation ... if `alloc` is nil ...